### ISSUE TITLE
fmriprep html not including all functional reports

### ISSUE THREAD

**Issue by jdkent** [poldracklab/fmriprep#1923]
Did some runs of fmriprep 1.5.4 and noticed that if I have some functional images that include the `run` identifier, but others don't, only the reports for the functional images that include the `run` identifier are included in the final html report.

Looking in the working directory, it appears all the `svg` files are being produced, but they are not being copied over to the derivatives figures directory.

I modified `ds005` to have two different tasks (one with a `run` identifier and one without) and ran the following fmriprep command:

```
fmriprep-docker -i poldracklab/fmriprep:1.5.4 -w ${PWD}/work ${PWD}/ds005 ${PWD}/derivatives --sloppy --notrack --nthreads 3 --fs-no-reconall --fs-license-file ${PWD}/license.txt --output-spaces func -vv
```
[here is the data](https://osf.io/urjk4/download) including the modified ds005, the working directory and the derivatives output.

EDIT:
my first guess is the problem starts in niworkflows [where we are taking the combination of all available entities](https://github.com/poldracklab/niworkflows/blob/9a07072cc52351a0fe85648fa5e864f9b47192df/niworkflows/reports/core.py#L314) (not all inputs will have a value for all the entities)

**Comment by oesteban** [comment]
Could this be a result of this - https://neurostars.org/t/missing-runs-from-reports/4758/25 ?

I'm not against adding a fix, esp. because you added tests to support it, but is it possible that this allows for ambiguous queries?

**Comment by jdkent** [comment]
Perhaps I do not understand what an ambiguous query would look like. I could see a problem if the report did not take into account additional relevant entities like `acq`:
```
sub-01_task-faketask_acq-multiband_desc-aroma_bold.svg
sub-01_task-faketask_acq-singleband_desc-aroma_bold.svg
```
but that would be fixed by adding `acq` to orderings in [`fmriprep.yml`](https://github.com/poldracklab/niworkflows/blob/24619661af90cd40bf921e6d8de08027a6ae76fc/niworkflows/reports/fmriprep.yml#L36), correct?

Or is my example off base from what you are suggesting?

**Comment by oesteban** [comment]
> Or is my example off base from what you are suggesting?

I don't think so. If thinking of an ambiguous example is so hard, then this solution seems much better than recommending the user to define always run when one of the tasks requires to have them. I guess, the only ambiguity arises when one utilizes that to allocate more files:

```
sub-01_task-faketask_bold.nii.gz
sub-01_task-faketask_run-1_bold.nii.gz
sub-01_task-faketask_run-2_bold.nii.gz
```
which I don't know for sure whether it is explicitly prohibited by BIDS in the first place. But one could see the run without `run` that it indeed corresponds to a `run-0` kind of thing.

**Comment by jdkent** [comment]
I think this can be closed with #1937
